WS2012R2/lisa/setupscripts/Cleanup_Backup_Disk.ps1

```diff
@@ -0,0 +1,99 @@
+########################################################################
+#
+# Linux on Hyper-V and Azure Test Code, ver. 1.0.0
+# Copyright (c) Microsoft Corporation
+#
+# All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the ""License"");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+# ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR
+# PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+#
+# See the Apache Version 2.0 License for specific language governing
+# permissions and limitations under the License.
+#
+########################################################################
+
+
+<#
+  .Synopsis
+    This script creates a disk to be used for backup and restore tests
+  .Description
+    This script will create a new VHD double the size of the VHD in the
+    given vm. The VHD will be mounted to a new partiton, initialized and
+    formatted with NTFS
+
+    Note: The script has to be run on the host.
+
+  .Parameter vmName
+    Name of the VM that will be tested and for which the VHD will be
+    deleted
+#>
+
+param ([string]$vmName)
+
+$retVal = $False
+
+#############################################################
+#
+# Main script body
+#
+#############################################################
+
+# Check input arguments
+if ($vmName -eq $null)
+{
+    "ERROR: VM name is null"
+    return $retVal
+}
+
+$summaryLog = "${vmName}_summary.log"
+
+# Source STOR_VSS_Utils.ps1 for common VSS functions
+if (Test-Path ".\setupScripts\STOR_VSS_Utils.ps1") {
+  . .\setupScripts\STOR_VSS_Utils.ps1
+  Write-Output "Sourced STOR_VSS_Utils.ps1" | Tee-Object -Append -file $summaryLog
+}
+else {
+  Write-Output "Could not find setupScripts\STOR_VSS_Utils.ps1" | Tee-Object -Append -file $summaryLog
+    return $false
+}
+
+$backupdiskpath = (get-vmhost).VirtualHardDiskPath + "\" + $vmName + "_VSS_DISK.vhdx"
+$tempFile = (get-vmhost).VirtualHardDiskPath + "\" + $vmName + "_DRIVE_LETTER.txt"
+
+$var = getDriveLetter $vmName
```

Comment:
The drive letter is set by reading it from the temporary txt file the setup script creates. Given that the function that does the reading resides inside the STOR_VSS_Utils script the issues is probably related to the one you mentioned above. The missing "cd $rootdir" command in Setup and Cleanup scripts. This will probably be solved by solving the above issues.